source/Octopus.Versioning/Octopus/OctopusVersionMask.cs

```diff
@@ -177,7 +177,7 @@ public override bool IsSubstitute
                     if (!IsPresent)
                         return false;
 
-                    return Regex.IsMatch(matchGroup.Value, @$"{OctopusVersionMaskParser.PatternIncrement}|{OctopusVersionMaskParser.PatternCurrent}$");
+                    return Regex.IsMatch(matchGroup.Value, @$"\.({OctopusVersionMaskParser.PatternIncrement}|{OctopusVersionMaskParser.PatternCurrent})$");
```

Comment:
```suggestion
                    return Regex.IsMatch(matchGroup.Value, @$"\.(?:{OctopusVersionMaskParser.PatternIncrement}|{OctopusVersionMaskParser.PatternCurrent})$");
```

If not capturing a group, you should explicitly mark it as non-capturing.

This was the result of a simple test I ran against this code:

|       Method |     Mean |    Error |  StdDev |
|------------- |---------:|---------:|--------:|
|    Capturing | 372.6 ns | 18.68 ns | 1.02 ns |
| NonCapturing | 335.9 ns | 12.76 ns | 0.70 ns |

Trivial for this example, but a good practice nevertheless.

Reply:
👍 I didn't realise that had a performance benefit. I'll add that change.